src/main/java/kitchenpos/menu/domain/Menu.java

```diff
@@ -0,0 +1,92 @@
+package kitchenpos.menu.domain;
+
+import kitchenpos.wrap.Price;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+
+@Entity
+public class Menu {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String name;
+
+    @Embedded
+    private Price price;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "menu_group_id", foreignKey = @ForeignKey(name = "fk_menu_menu_group"))
+    private MenuGroup menuGroup;
+
+    @Embedded
+    private MenuProducts menuProducts = new MenuProducts();
+
+    protected Menu() {
+    }
+
+    public Menu(Long id){
+        this.id = id;
+    }
+
+    public Menu(Long id, String name, Price price, MenuGroup menuGroup, MenuProducts menuProducts) {
+        priceValidate(price, menuProducts);
+        this.id = id;
+        this.name = name;
+        this.price = price;
+        this.menuGroup = menuGroup;
+        this.menuProducts = menuProducts;
+        menuProducts.registerProduct(this);
+    }
+
+    public Menu(String name, Price price, MenuGroup menuGroup, MenuProducts menuProducts) {
+        priceValidate(price, menuProducts);
+        this.name = name;
+        this.price = price;
+        this.menuGroup = menuGroup;
+        this.menuProducts = menuProducts;
+        menuProducts.registerProduct(this);
+    }
+
+    private void priceValidate(Price price, MenuProducts menuProducts) {
```

Comment:
```suggestion
    private void validatePrice(Price price, MenuProducts menuProducts) {
```
메서드니까 동사를 먼저 써주는게 덜 어색하겠네요

Reply:
앗,,,
수정했습니다!!
😀